core/primitives/src/types.rs

```diff
@@ -558,6 +558,12 @@ pub enum TransactionOrReceiptId {
     Receipt { receipt_id: CryptoHash, receiver_id: AccountId },
 }
 
+/// Cache for compiled modules
+pub trait CompiledContractCache: Send + Sync {
+    fn put(&self, key: &[u8], value: &[u8]) -> Result<(), std::io::Error>;
+    fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, std::io::Error>;
```

Comment:
Can `key` be a `CryptoHash`?

Reply:
Good question. Guess we wouldn't win much from that, and probably could use more complex key structure. Let's keep it as is for now.